src/pages/Locations/LocationItem.tsx

```diff
@@ -0,0 +1,55 @@
+import type { FC } from 'react';
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { IonList, IonItem, IonText } from '@ionic/react';
+import { Loader } from 'src/components';
+import './LocationItem.css';
+import SearchBar from '../../components/SearchBar';
+import type { Location } from '../../types';
+import type { UseLocationResultProps } from './hooks/LocationsProps';
+
+const LocationItem: FC<UseLocationResultProps> = ({ groupedLocations, loading }) => {
+  const { t } = useTranslation();
+  const [locationResult, setLocationResult] = useState<Location[]>([]);
+
+  const handleSearch = (searchText: string) => {
+    const regex = new RegExp(searchText, 'i');
+    const filteredResults: Location[] = groupedLocations.filter((location) => regex.test(location.name));
+    setLocationResult(filteredResults);
+  };
+
+  useEffect(() => {
+    if (!loading) setLocationResult(groupedLocations);
+  }, [loading, groupedLocations]);
+
+  return (
+    <>
+      <SearchBar onSearch={handleSearch} />
+      {loading ? (
+        <Loader />
+      ) : locationResult.length > 0 ? (
+        locationResult.map((eachLocation: Location) => {
+          return (
+            <IonList lines="none" key={eachLocation.uuid} className="custom-list">
+              <IonItem className="custom-item" detail={false} routerLink={`/locations/${eachLocation.uuid}`} button>
```

Comment:
router link doesn't use the link from route urls mwa-reference/src/utils/constants/router.ts